app/models/concerns/smart_proxy_reference.rb

```diff
@@ -0,0 +1,51 @@
+module SmartProxyReference
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    attr_accessor :smart_proxy_references
+
+    def smart_proxy_reference(hash)
+      raise Foreman::Exception.new(N_("Proxy reference should be { :relation => [:column_name, ...]}")) unless correct_reference_format?(hash)
+      @smart_proxy_references ||= {}
+      @smart_proxy_references.merge!(hash)
```

Comment:
This code prevents adding fields from the same relation in different statements:
Consider:
``` ruby
smart_proxy_reference :subnets => [:dhcp_id, :dns_id]
#then later in code (maybe in a different plugin)
smart_proxy_reference :subnets => [:tftp_id]
```
`:dhcp_id` and `:dns_id` would be removed, since `merge` doesn't deal with array values.